### PR TITLE
Expose streams in public quantile APIs

### PR DESCRIPTION
**Pull request by shrshi**
## Description
Adds stream parameter to 
```
cudf::quantile
cudf::quantiles
cudf::percentile_approx
```
Added stream gtests to verify correct stream forwarding.

Reference: #13744 

## Checklist
- [X] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [X] New or existing tests cover these changes.
- [X] The documentation is up to date with these changes.

### CODE REVIEW

File: cpp/tests/streams/quantile_test.cpp [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+  auto input = cudf::table_view({input_a, input_b});
+
+  auto actual = cudf::quantiles(input,
+                                {0.0f, 0.5f, 0.7f, 0.25f, 1.0f},
+                                cudf::interpolation::NEAREST,
+                                cudf::sorted::NO,
+                                {cudf::order::ASCENDING, cudf::order::DESCENDING},
+                                {},
+                                cudf::test::get_default_stream());
```
Comment (pmattione-nvidia) [inline cpp/tests/streams/quantile_test.cpp:52]: I would add some CUDF_TEST_EXPECT_TABLES_EQUAL (or whichever is appropriate) checks to these tests.

Comment (shrshi) [reply cpp/tests/streams/quantile_test.cpp:52]: Stream tests only verify if the stream is being forwarded correctly. We do not check functional correctness here; the test tracks the test stream passed and verifies that it is used correctly for all operations.

File: cpp/tests/streams/quantile_test.cpp [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cudf_test/base_fixture.hpp>
+#include <cudf_test/column_wrapper.hpp>
+#include <cudf_test/table_utilities.hpp>
+#include <cudf_test/type_lists.hpp>
```
Comment (davidwendt) [inline cpp/tests/streams/quantile_test.cpp:20]:
```suggestion
```
Not sure if this are needed.

File: cpp/include/cudf/quantiles.hpp [hunk 3]
```diff
@@ -114,6 +118,7 @@ std::unique_ptr<table> quantiles(
  *
  * @param input           tdigest input data. One tdigest per row
  * @param percentiles     Desired percentiles in range [0, 1]
+ * @param stream CUDA stream used for device memory operations and kernel launches
  * @param mr              Device memory resource used to allocate the returned column's device
  * memory
```
Comment (davidwendt) [inline cpp/include/cudf/quantiles.hpp:123]:
```suggestion
 * @param input One tdigest per row
 * @param percentiles Desired percentiles in range [0, 1]
 * @param stream CUDA stream used for device memory operations and kernel launches
 * @param mr Device memory resource used to allocate the returned column's device memory
```
Could you fix the formatting of all of these?

**Review by davidwendt — CHANGES_REQUESTED** [review]

File: cpp/tests/streams/quantile_test.cpp (hunk 1 above)
Comment (pmattione-nvidia) [reply cpp/tests/streams/quantile_test.cpp:52]: Naively, this code looks like it's intending to test whether these cudf functions (e.g. quantiles()) work properly.  If we don't want to test that here, please add small comments to that effect.

**Review by pmattione-nvidia — APPROVED** [review]

**Review by davidwendt — APPROVED** [review]

**Pull request merged**